metaseq/tasks/streaming_language_modeling.py

```diff
@@ -51,6 +51,10 @@
 LANGUAGE_MODELING_MODE = ChoiceEnum(["standard", "cm3", "racm3"])
 CM3_MODE = ChoiceEnum(["poisson", "fixed", "fim"])
 
+DatasetWithShardInformation = namedtuple(
+    "DatasetWithShardInformation", ["dataset", "is_sharded", "shard_id", "num_shards"]
```

Comment:
where exactly are we using this extra fields ? for eg - num_shards : in places like get_batch_iterator its passed as parameter?

Reply:
i saw dataset.is_sharded being used

Reply:
is it possible to add some flag to jsonl dataset itself and remove this named tuple completely?

Reply:
We pass JsonlDataset into ConcatDataset. I would rather have a wrapper over a dataset rather than have to propagate this information through all datasets. WDYT?

Reply:
so is ("shard_id", "num_shards") used somewhere else?

Reply:
added extra assertions using this to insure consistency between iterator/dataset